pkg/controller/jobcontrol.go

```diff
@@ -214,10 +232,21 @@ func (c *jobControl) ControlJobs(
 		return JobControlDeletingJobs, nil, err
 	}
 
+	// The owner needs a job to process its current generaton. A job needs to
+	// be created to do this processing.
 	if generationJob != nil {
+
+		logger.WithFields(log.Fields{
+			"generationJob": generationJob.Name,
+		}).Debugf("generationJob")
+
 		switch {
 		// A successful job exists for the current generation of the owner.
 		// The owner's status needs to be updated accordingly.
+		case buildReprocessingJob && (isSuccessful(generationJob) || isFailed(generationJob)):
+			logger.Debugf("successful generation job found but reprocess job requested, building new job")
+			_, err := c.createJob(ownerKey, owner, jobFactory, logger)
+			return JobControlCreatingJob, nil, err
```

Comment:
DoesOwnerNeedReprocessing and the subsequent block here were added  because lumping the reprocess request into the pre-existing DoesOwnerNeedProcessing, there's no way to tell when a job is successful the first time and we *don't* want to immediately re-run for config management. Both states seem necessary to get the proper behavior and not duplicate/infinite jobs.